model/game/tournament.js

```diff
@@ -3,27 +3,39 @@ let Match = require("./match");
 let _ = require("lodash");
 
 class Tournament {
-  constructor(strategies, numberOfMatches) {
+  constructor(strategies, numberOfMatches, gamesPerMatch) {
     this.strategies = strategies;
     let num = 0;
-    this.players = _.map(strategies, strategy => new Player(num++, strategy));
+    this.players = _.map(strategies, strategy => new Player(strategy.name, strategy.strategy));
     this.numberOfMatches = numberOfMatches;
+    this.gamesPerMatch = gamesPerMatch;
     this.matches = [];
   }
 
   play() {
-    _.times(this.numberOfMatches, () => this.matches.push(new Match(this.players, 4)));
-    _.forEach(this.matches, match => match.play());
+    let promise = Promise.resolve();
+    _.times(this.numberOfMatches, () => this.matches.push(new Match(this.players, this.gamesPerMatch)));
+
+    // I think if we deep-clone the players once it would allow for doing this concurrently,
+    // and I could used Promise.all([promises]) to wait for them, but for now chain the promises.
+    _.forEach(this.matches, match => promise = promise.then(() => match.play()));
+    return promise;
   }
 
   listWinners() {
     return _.map(this.matches, match => match.winner());
   }
 
   report() {
-    return _.map(_.groupBy(this.listWinners()), (value, key) => {
-      return { playerId: key, wins: value.length / this.numberOfMatches };
+    let report = _.map(_.groupBy(this.listWinners()), (value, key) => {
```

Comment:
I think it will be interesting.  We'll probably want to know if the matches are close.